### ISSUE TITLE
Interleaving logs

### ISSUE THREAD

**Issue by Limess** [viesti/timbre-json-appender#19]
After running this library in production on a system with a high log volume we're seeing some weird interleaving of log lines.

This is probably as `println` is not safe from interleaving logs across threads out-of-the-box.

See this fix for the timbre println appender: https://github.com/ptaoussanis/timbre/commit/057b5a4c871752957e50c3eaf667c0517d56ca9a

**Comment by viesti** [comment]
Ah true, thanks reporting! A similar fix can be done here too.

**Comment by viesti** [comment]
Pushed out 0.2.2 with the fix (c8bcd46a3ecc98dec9272c550f8f9cc89fdbef85) like in timbre, tested it out like this:

```
user=> (pmap #(timbre/info %) (range 10))
{"msg":6,"timestamp":"2021-07-31T09:36:39Z","level":"info","thread":"clojure-agent-send-off-pool-2","hostname":"paju.local"}
{"msg":7,"timestamp":"2021-07-31T09:36:39Z","level":"info","thread":"clojure-agent-send-off-pool-7","hostname":"paju.local"}{"msg":8,"timestamp":"2021-07-31T09:36:39Z","level":"info","thread":"clojure-agent-send-off-pool-8","hostname":"paju.local"}
{"msg":4,"timestamp":"2021-07-31T09:36:39Z","level":"info","thread":"clojure-agent-send-off-pool-0","hostname":"paju.local"}
{"msg":2,"timestamp":"2021-07-31T09:36:39Z","level":"info","thread":"clojure-agent-send-off-pool-5","hostname":"paju.local"}
{"msg":9,"timestamp":"2021-07-31T09:36:39Z","level":"info","thread":"clojure-agent-send-off-pool-3","hostname":"paju.local"}
{"msg":0,"timestamp":"2021-07-31T09:36:39Z","level":"info","thread":"clojure-agent-send-off-pool-6","hostname":"paju.local"}
{"msg":5,"timestamp":"2021-07-31T09:36:39Z","level":"info","thread":"clojure-agent-send-off-pool-10","hostname":"paju.local"}

{"msg":3,"timestamp":"2021-07-31T09:36:39Z","level":"info","thread":"clojure-agent-send-off-pool-4","hostname":"paju.local"}{"msg":1,"timestamp":"2021-07-31T09:36:39Z","level":"info","thread":"clojure-agent-send-off-pool-9","hostname":"paju.local"}
(nil nil nil nil nil nil nil nil nil nil)
user=> (require '[timbre-json-appender.core :as tas] :reload) ;; Fix applied
nil
user=> (tas/install)
{:min-level :info, :appenders {:json {:enabled? true, :async? false, :min-level nil, :fn #function[timbre-json-appender.core/json-appender/fn--11122]}}}
user=> (pmap #(timbre/info %) (range 10))
{"msg":8,"timestamp":"2021-07-31T09:37:03Z","level":"info","thread":"clojure-agent-send-off-pool-0","hostname":"paju.local"}
{"msg":1,"timestamp":"2021-07-31T09:37:03Z","level":"info","thread":"clojure-agent-send-off-pool-6","hostname":"paju.local"}
{"msg":3,"timestamp":"2021-07-31T09:37:03Z","level":"info","thread":"clojure-agent-send-off-pool-7","hostname":"paju.local"}
{"msg":4,"timestamp":"2021-07-31T09:37:03Z","level":"info","thread":"clojure-agent-send-off-pool-9","hostname":"paju.local"}
{"msg":9,"timestamp":"2021-07-31T09:37:03Z","level":"info","thread":"clojure-agent-send-off-pool-2","hostname":"paju.local"}
{"msg":7,"timestamp":"2021-07-31T09:37:03Z","level":"info","thread":"clojure-agent-send-off-pool-3","hostname":"paju.local"}
{"msg":6,"timestamp":"2021-07-31T09:37:03Z","level":"info","thread":"clojure-agent-send-off-pool-10","hostname":"paju.local"}
{"msg":2,"timestamp":"2021-07-31T09:37:03Z","level":"info","thread":"clojure-agent-send-off-pool-4","hostname":"paju.local"}
{"msg":0,"timestamp":"2021-07-31T09:37:03Z","level":"info","thread":"clojure-agent-send-off-pool-5","hostname":"paju.local"}
{"msg":5,"timestamp":"2021-07-31T09:37:03Z","level":"info","thread":"clojure-agent-send-off-pool-8","hostname":"paju.local"}
(nil nil nil nil nil nil nil nil nil nil)
```

Thanks again! :)
